### PR TITLE
Documentation: Improve link layout, add getting started doc

### PR DESCRIPTION
**Pull request by derekparker**
Improve the link ordering for the main README and add a
"Getting Started" doc with basic usage information for new users.

### CODE REVIEW

File: Documentation/cli/getting_started.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+├── cmd
+│   └── foo
+│       └── main.go
+├── pkg
+│   └── baz
+│       ├── bar.go
+│       └── bar_test.go
+```
+
+If you are in the directory `github.com/me/foo` you can simple run `dlv debug`
```
Comment (aarzilli) [inline Documentation/cli/getting_started.md]: I think here you meant `github.com/me/foo/cmd/foo`

File: Documentation/cli/getting_started.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+from the command line. From anywhere else, say the project root, you can simply
+provide the package: `dlv debug github.com/me/foo`.
+
+Invoking that command will cause Delve to compile the program is a way most
```
Comment (aarzilli) [inline Documentation/cli/getting_started.md]: s/is a way/in a way/

File: Documentation/cli/getting_started.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+suitable for debugging, then it will execute and attach to the program and begin
+a debug session. Now, when the debug session has first started you are at the
+very beginning of the programs initialization. To get to someplace more useful
```
Comment (aarzilli) [inline Documentation/cli/getting_started.md]: s/programs/program's/

File: Documentation/cli/getting_started.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+you're going to want to set a breakpoint or two and continue execution to that
+point.
+
+For example, to continue execution to your programs `main` function:
+
+```
+$ dlv debug github.com/me/foo
```
Comment (aarzilli) [inline Documentation/cli/getting_started.md]: Should be github.com/me/foo/cmd/foo

File: Documentation/cli/getting_started.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+## Debugging tests
+
+Given the same directory structure as above you can debug your code by executing
+your test suite. For this you can use the `dlv test` subcommand, which takes the
+same optional package path as `dlv debug`, and will also build the current
+package if not given any argument.
+
+```
+$ dlv test github.com/pkg/baz
```
Comment (aarzilli) [inline Documentation/cli/getting_started.md]: github.com/me/foo/pkg/baz

File: Documentation/cli/getting_started.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+Type 'help' for list of commands.
+(dlv) funcs test.Test*
+_/home/deparker/code/scratch/golang/test.TestHi
```
Comment (aarzilli) [inline Documentation/cli/getting_started.md]: I think this should be changed to match the directory structure in the example.

File: Documentation/cli/getting_started.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+     4:	
+=>   5:	func TestHi(t *testing.T) {
+     6:		t.Fatal("implement me!")
+     7:	}
+(dlv) 
+```
+
+As you can see, we began debugging the test binary, found our test function via
+the `funcs` command which takes a regexp to filter the list of functions, set a
+breakpoint and then continued execution until we hit that breakpoint.
```
Comment (aarzilli) [inline Documentation/cli/getting_started.md]: I think this should mention that `break` followed with just the function name will work if the function name is unambiguous. I worry that people may read this and get the impression that the only way to set a breakpoint is the full function path, which is inconvenient.

**Comment by derekparker** [comment]
@aarzilli updated.

**Comment by zchee** [comment]
@aarzilli As @navossoc said, `Getting Started` is a direct link to markdown. So need `.md` suffix to link url.

ref: https://github.com/derekparker/delve/commit/01f5b3c8d0a2f2ee43cc327b3a6d6b7b7ae53ae5#commitcomment-26638020

**Comment by zchee** [comment]
Ah, this pr written by @derekparker.
Also cc: @derekparker.

**Pull request merged**